Library/ViewModels/RewardsCollectionViewModel.swift

```diff
@@ -27,6 +31,17 @@ RewardsCollectionViewModelInputs, RewardsCollectionViewModelOutputs {
     )
       .map(first)
       .map { $0.rewards }
+
+    let selectedReward = reloadDataWithRewards
+      .takePairWhen(self.rewardSelectedIndexProperty.signal.skipNil())
+      .map { rewards, index in rewards[index] }
```

Comment:
Since the `dataSource` is populated with rewards, I'd say it's highly unlikely that we would ever have an out of index exception. I'd err on the side of "let's keep an eye on it" and address the underlying issue if it comes up, since that would be a symptom of a more serious bug somewhere else in this view model if that were to happen.